src/app/pages/categories/management/pages-categories-management.component.spec.ts

```diff
@@ -0,0 +1,170 @@
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(PagesCategoriesManagementComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  describe('ngOnInit', () => {
+    describe('get transaction categories', () => {
+      describe('success', () => {
+        describe('empty table', () => {
+          it('should not display table', fakeAsync(() => {
+            categoriesSubject.next([]);
+            flushMicrotasks();
+            fixture.detectChanges();
+
+            const table = fixture.debugElement.query(
+              By.css('.mat-table'),
+            );
+
+            expect(table).toBeFalsy();
+          }));
+        });
+
+        describe('not empty table', () => {
+          it('should display categories in table', fakeAsync(() => {
+            categoriesSubject.next(transactionCategoriesObjectsMockFunc());
+            flushMicrotasks();
+            fixture.detectChanges();
+
+            const table = fixture.debugElement.query(
+              By.css('.mat-table'),
+            ).nativeElement as HTMLDivElement;
+
+            const sortecCategories = sortAlphabeticallyByProp<TransactionCategory, 'name'>
+            (transactionCategoriesObjectsMockFunc(), 'name');
+
+            expect(table).toBeTruthy();
+            expect(component.transactionCategoriesData.hasError).toBeFalse();
+            expect(component.transactionCategoriesData.data).toEqual(sortecCategories);
+          }));
+        });
+      });
+
+      describe('error', () => {
+        it('should set TStateData.error as true and data as null', fakeAsync(() => {
+          categoriesSubject.error('Some error');
+          flushMicrotasks();
+
+          expect(component.transactionCategoriesData.hasError).toBeTrue();
+          expect(component.transactionCategoriesData.data).toEqual(null);
+        }));
+
+        it('should display error wall', fakeAsync(() => {
+          categoriesSubject.error('Some error');
+          flushMicrotasks();
+          fixture.detectChanges();
+
+          const messageDiv = fixture.debugElement.query(
+            By.css('.message-container'),
+          ).nativeElement as HTMLDivElement;
+
+          expect(messageDiv).toBeTruthy();
+        }));
+      });
+    });
+  });
+
+  describe('filter transactions', () => {
+    let matSelect: HTMLDivElement;
+
+    beforeEach( fakeAsync(() => {
+      categoriesSubject.next(transactionCategoriesObjectsMockFunc());
+      flushMicrotasks();
+      fixture.detectChanges();
+
+      matSelect = fixture.debugElement.query(
+        By.css('.mat-select-trigger'),
+      ).nativeElement as HTMLDivElement;
+
+      matSelect.click();
+      fixture.detectChanges();
+    }));
+
+    describe('filter categories by Expenses', () => {
+      it('selected Expenses on select-box should filter categories with expense type', fakeAsync( async () => {
```

Comment:
Seems like this is randomly failing when running `npm run test:headless`. Seems like adding:
```
discardPeriodicTasks();
flush();
```
after last `expect` in tests fixes the issues. Please do it for both Incomes and Expenses tests